src/overlays/actors/ovl_Bg_Gnd_Darkmeiro/z_bg_gnd_darkmeiro.c

```diff
@@ -35,25 +33,189 @@ const ActorInit Bg_Gnd_Darkmeiro_InitVars = {
+        if (this->actionFlags & 4) {
+            if (this->timer1 > 0) {
+                this->timer1--;
+            } else {
+                Flags_UnsetSwitch(globalCtx, ((this->dyna.actor.params >> 8) & 0x3F) + 1);
+                this->actionFlags &= ~4;
+            }
+        } else {
+            this->actionFlags |= 4;
+            this->timer1 = 304;
+            Audio_PlaySoundGeneral(NA_SE_EV_RED_EYE, &D_801333D4, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+        }
+    }
+
+    if (Flags_GetSwitch(globalCtx, ((this->dyna.actor.params >> 8) & 0x3F) + 2)) {
+        if (this->actionFlags & 8) {
+            if (this->timer2 > 0) {
+                this->timer2--;
+            } else {
+                Flags_UnsetSwitch(globalCtx, ((this->dyna.actor.params >> 8) & 0x3F) + 2);
+                this->actionFlags &= ~8;
+            }
+        } else {
+            this->actionFlags |= 8;
+            this->timer2 = 304;
+            Audio_PlaySoundGeneral(NA_SE_EV_RED_EYE, &D_801333D4, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+        }
+    }
+
+    timeLeft = CLAMP_MIN(this->timer1, this->timer2);
+    if (timeLeft > 0) {
+        func_8002F994(&this->dyna.actor, timeLeft);
+    }
+    if ((this->timer1 >= 64) || (this->timer2 >= 64)) {
+        Flags_SetSwitch(globalCtx, (this->dyna.actor.params >> 8) & 0x3F);
+    } else {
+        Flags_UnsetSwitch(globalCtx, (this->dyna.actor.params >> 8) & 0x3F);
+    }
+}
+
+void BgGndDarkmeiro_UpdateStaticBlock(BgGndDarkmeiro* this, GlobalContext* globalCtx) {
+}
+
+void BgGndDarkmeiro_UpdateSwitchBlock(BgGndDarkmeiro* this, GlobalContext* globalCtx) {
+    if (this->timer1 > 0) {
+        this->timer1--;
+    }
+
+    if (Flags_GetSwitch(globalCtx, (this->dyna.actor.params >> 8) & 0x3F)) {
+        this->timer1 = 64;
+    }
+
+    BgGndDarkmeiro_ToggleBlock(this, globalCtx);
+}
+
+void BgGndDarkmeiro_Update(Actor* thisx, GlobalContext* globalCtx) {
+    BgGndDarkmeiro* this = THIS;
+    GlobalContext* globalCtx2 = globalCtx;
+
+    this->updateFunc(this, globalCtx2);
+}
+
+void BgGndDarkmeiro_DrawInvisiblePath(Actor* thisx, GlobalContext* globalCtx) {
+    Gfx_DrawDListXlu(globalCtx, D_060088B0);
+}
+
+void BgGndDarkmeiro_DrawSwitchBlock(Actor* thisx, GlobalContext* globalCtx) {
+    BgGndDarkmeiro* this = THIS;
+    s16 vanishTimer;
+
+    vanishTimer = this->timer1;
+    if (vanishTimer != 0) {
+        if (vanishTimer > 64) {
+            this->timer2 = (this->timer2 < 120) ? this->timer2 + 8 : 127;
+        } else if (vanishTimer > 16) {
+            this->timer2 = (Math_Coss((u16)this->timer1 * 0x1000) * 64.0f) + 127.0f;
+            if (this->timer2 > 127) {
+                this->timer2 = 127;
+            }
+        } else {
+            this->timer2 = vanishTimer * 8;
+        }
+
+        OPEN_DISPS(globalCtx->state.gfxCtx, "../z_bg_gnd_darkmeiro.c", 378);
+        //@ bug: Due to a bug in the display list, the transparency data is not used.
```

Comment:
The syntax needs to be fixed here if we want the bug annotation to be detected properly
```suggestion
        // @bug Due to a bug in the display list, the transparency data is not used.
```